### ISSUE TITLE
[BUG] Could not find a declaration file for module '@playwright/test' with module NodeNext

### ISSUE THREAD

**Issue by PabloLION** [microsoft/playwright#14971]
## Context

- Playwright Version: 1.22.2
- Operating System: Mac with M1 chip
- Node.js version: 16.x-18.x
- Extra: TypeScript 4.7.4 with `"module": "NodeNext"`

### Code Snippet

```typescript
import { test, expect } from "@playwright/test";
```

## Describe the bug

- Repeating unresolved issue https://github.com/microsoft/playwright/issues/9819 with [MRE](https://github.com/PabloLION/issue-playwright-yarn-declaration)
- Typescript 4.7 has been released with a better ESM support. But with `"module": "NodeNext"`, my ts engine throws and `ts(7016)` error on `@playwright/test` installed via `yarn`:

  ```
  Could not find a declaration file for module '@playwright/test'. '~/issue-playwright-yarn-declaration/node_modules/@playwright/test/index.mjs' implicitly has an 'any' type.
    Try `npm i --save-dev @types/playwright__test` if it exists or add a new declaration (.d.ts) file containing `declare module '@playwright/test';`ts(7016)
  ```
- I've set up a MRE with the steps in [Repro](#repro) chapter

## Repro

1. yarn init -y
2. yarn add @playwright/test@latest typescript@latest
3. npx tsc --init
4. add an example as the [TS example](https://playwright.dev/docs/intro#first-test)
5. change "tsconfig.compilerOptions.module" to "NodeNext".
6. add a "playwright.config.ts" from [TS config](https://playwright.dev/docs/intro#configuration-file)

The error occurs on the step(5.). However, the test can run despite that TS engine is throwing error, also the ESLint is giving out a bunch of ``Unsafe call of an `any` typed value`` with the MRE. This breaks the git-hook and I cannot find an easy way out.

## Work around

I've found one work around giving up all TS and ESLint-typescript features:

- add `"src/test/*"` to the `${eslint_Config}.ignorePatterns`.
- add `./src/test/**/*` to the `tsconfig.exclude`.

**Comment by PabloLION** [comment]
Hi, @mxschmitt. It seems to me that you have some experience with this issue. Would you mind self-assign to this to save some works for the team?

**Comment by mxschmitt** [comment]
This should be fixed already. Try: `yarn add @playwright/test@next.

**Comment by mxschmitt** [comment]
Please file a new issue if you still encounter issues.

**Comment by PabloLION** [comment]
Tested that the `1.23.0` fixes this issue. Thanks.

TL;DR
Sorry for the inactivity.
Test detail: I've tested with the same MRE https://github.com/PabloLION/issue-playwright-yarn-declaration.
